core/src/main/java/com/linecorp/armeria/client/websocket/WebSocketSession.java

```diff
@@ -123,9 +128,29 @@ public void setOutbound(Publisher<? extends WebSocketFrame> outbound) {
             streamMessage = new PublisherBasedStreamMessage<>(outbound);
         }
 
-        if (!outboundFuture.complete(
-                streamMessage.map(webSocketFrame -> HttpData.wrap(encoder.encode(ctx, webSocketFrame))))) {
-            streamMessage.abort();
+        final StreamMessage<HttpData> data =
+                streamMessage.map(webSocketFrame -> HttpData.wrap(encoder.encode(ctx, webSocketFrame)))
+                        .recoverAndResume(cause -> {
+                            if (cause instanceof ClosedStreamException) {
+                                return StreamMessage.aborted(cause);
+                            }
+                            ctx.logBuilder().endRequest(cause);
+                            ctx.logBuilder().endResponse(cause);
```

Comment:
Question) I'd rather the log isn't ended here since there may be additional data written to `RequestLogBuilder` (`responseHeaders`, etc..).
What do you think of removing this call?

```suggestion
```